go/instance.go

```diff
@@ -29,15 +29,17 @@ type Instance[T signature.Signature] struct {
 	runtimeCtx signature.RuntimeContext
 }
 
-func (r *Runtime[T]) Instance(next Next[T]) (*Instance[T], error) {
+func (r *Runtime[T]) Instance(next ...Next[T]) (*Instance[T], error) {
+	ctx := r.new()
 	i := &Instance[T]{
-		next:       next,
 		runtime:    r,
-		ctx:        r.signature,
-		runtimeCtx: r.signature.RuntimeContext(),
+		ctx:        ctx,
+		runtimeCtx: ctx.RuntimeContext(),
 	}
 
-	if i.next == nil {
+	if len(next) > 0 {
```

Comment:
Is providing more than 1 next an error? eg should we restrict it to none or one only?

Reply:
Not an error but we expect only one - if users want to pass in more than one next function we'll use only the first one and the subsequent ones will be ignored. 

The given next function is totally allowed to call out to another next function and that's the main reason we won't allow just passing in multiple next functions.